app/controllers/comments_controller.rb

```diff
@@ -34,17 +34,21 @@ def create
           next if @target.user_id == duid
 
           Notification.create(body: "#{poster_name} has also commented on submission #{@target.display_title} (ID #{@target.id}).",
-                              source_type: "Submission",
-                              source_id: @target.id,
+                              source_type: 'Comment',
+                              source_id: @comment.id,
                               user_id: duid,
                               url: submission_path(@target.id))
         end
         # Send a notification to the artist if it's not ourself.
-        Notification.create(body: "#{poster_name} has commented on your submission #{@target.display_title} (ID #{@target.id}).",
-                            source_type: "Submission",
-                            source_id: @target.id,
-                            user_id: @target.user_id,
-                            url: submission_path(@target.id)) unless @target.user_id == current_user.id
+        unless @target.user_id == current_user.id
+          Notification.create(
+            body: "#{poster_name} has commented on your submission #{@target.display_title} (ID #{@target.id}).",
+            source_type: 'Comment',
+            source_id: @comment.id,
+            user_id: @target.user_id,
+            url: @comment.url
+          )
+        end
```

Comment:
Interesting. Yeah, the source should be the comment itself. In the future, when watch functionality is implemented, we'll have Submission as the source.